pkg/networkserver/grpc_gsns.go

```diff
@@ -1166,7 +1166,11 @@ func (ns *NetworkServer) handleJoinRequest(ctx context.Context, up *ttnpb.Uplink
 
 	if !matched.SupportsJoin {
 		log.FromContext(ctx).Warn("ABP device sent a join-request, drop")
-		queuedEvents = append(queuedEvents, evtDropJoinRequest.NewWithIdentifiersAndData(ctx, matched.Ids, errABPJoinRequest))
+		queuedEvents = append(queuedEvents, evtDropJoinRequest.NewWithIdentifiersAndData(ctx, matched.Ids, errABPJoinRequest.New()))
+		return nil
+	}
+	if !mac.DeviceScheduleDownlinks(matched, ns.defaultMACSettings) {
```

Comment:
Since we're dropping the join request so early, we will not consume the `nonce`. For me this seems safe (it is equivalent to us not hearing the join request), but are there any other arguments against this ?